### PR TITLE
fix #163 - esbuild node_modules excludes

### LINKED ISSUE #163: Preview server error since 0.7.2 in yarn workspace
Have tried to update to `mailing@^0.7.2` and unfortunately I'm getting this error when attempting to run `mailing preview`.

```
Server Error
Error: Cannot find module 'http'

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Call Stack
webpackEmptyContext
file:///Users/willviles/Code/@Redbrain/packages/email/.mailing/.next/server/pages/index.js (22:10)
../../node_modules/axios/lib/adapters/http.js
webpack-internal:///./src/moduleManifest.js (1924:20)
__require2
webpack-internal:///./src/moduleManifest.js (34:56)
getDefaultAdapter
webpack-internal:///./src/moduleManifest.js (2240:27)
../../node_modules/axios/lib/defaults/index.js
webpack-internal:///./src/moduleManifest.js (2259:22)
__require2
webpack-internal:///./src/moduleManifest.js (34:56)
../../node_modules/axios/lib/core/transformData.js
webpack-internal:///./src/moduleManifest.js (2338:24)
__require2
webpack-internal:///./src/moduleManifest.js (34:56)
../../node_modules/axios/lib/core/dispatchRequest.js
webpack-internal:///./src/moduleManifest.js (2362:29)
__require2
webpack-internal:///./src/moduleManifest.js (34:56)
```

There are no errors in a vanilla project, however I'm using Mailing within a yarn workspace, which already contains a next.js app. 

Pinning to `mailing@0.7.1` and adding a the following nohoist rules to my root `package.json` ensures the error disappears:

```json
{
  "workspaces": {
    "nohoist": [
      "**/mailing",
      "**/mailing-core"
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by alexfarrill**
This fixes the problem in https://github.com/willviles/mailing-yarn-workspaces

### CODE REVIEW

File: packages/cli/src/commands/preview/server/setup.ts [hunk 1]
```diff
@@ -200,8 +199,11 @@ async function buildManifest(
     format: "esm",
     jsx: "preserve",
     external: [
-      ...Object.keys(pkg.dependencies || {}),
-      ...Object.keys(pkg.peerDependencies || {}),
+      "./node_modules/*",
```
Comment (alexfarrill) [inline packages/cli/src/commands/preview/server/setup.ts:202]: @psugihara this is a better way to do this.  any thoughts on best way to do this programmatically?  naive solution I could just prepend "../" for however deep cwd is, but there is probably a better way.  is there something you can check in `process`? i.e. ask node/npm, etc?

Comment (psugihara) [reply packages/cli/src/commands/preview/server/setup.ts:202]: this seems nice and simple

I think use `process.cwd().split("/").length` and just generate the string array like that.

This plugin looks in the filesystem but this way seems quicker if the string array works https://github.com/pradel/esbuild-node-externals/blob/main/esbuild-node-externals/src/utils.ts#L23

File: packages/cli/src/commands/preview/server/__test__/setup.test.ts [hunk 2]
```diff
@@ -25,101 +25,107 @@ function mockReadJSONSyncToThrowErrno() {
   });
 }
 
-describe("packageJsonVersionsMatch", () => {
-  it("should return false if .mailing package.json does not exist", async () => {
-    mockReadJSONSyncToThrowErrno();
-    const match = await packageJsonVersionsMatch();
-    expect(match).toBe(false);
+describe("setup", () => {
+  beforeEach(() => {
```
Comment (alexfarrill) [inline packages/cli/src/commands/preview/server/__test__/setup.test.ts:29]: this 3 liens is the only change that's not formatting

**Comment by alexfarrill** [comment]
e2e tests ✅

**Review by psugihara — APPROVED** [review]
looks great, nice fix!

**Review by psugihara — APPROVED** [review]

**Pull request merged**